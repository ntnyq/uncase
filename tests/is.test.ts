import { expect, it } from 'vitest'
import {
  isCamelCase,
  isCapitalCase,
  isConstantCase,
  isDotCase,
  isKebabCase,
  isNoCase,
  isPascalCase,
  isPascalSnakeCase,
  isPathCase,
  isSentenceCase,
  isSnakeCase,
  isTrainCase,
} from '../src/is'
import { allCaseValues } from './fixtures'

const CASE_VALUES_FIXTURES = [
  ...allCaseValues,
  // empty string is treated as false
  '',
]

it('isCamelCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isCamelCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isCapitalCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isCapitalCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isConstantCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isConstantCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isDotCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isDotCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isKebabCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isKebabCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isNoCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isNoCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isPascalCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isPascalCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isPascalSnakeCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isPascalSnakeCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isPathCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isPathCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isSentenceCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isSentenceCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isSnakeCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isSnakeCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})

it('isTrainCase', () => {
  const results = CASE_VALUES_FIXTURES.map(value => ({
    value,
    isValid: isTrainCase(value),
  }))

  expect(results.filter(v => v.isValid)).toHaveLength(1)
  expect(results).toMatchSnapshot()
})
