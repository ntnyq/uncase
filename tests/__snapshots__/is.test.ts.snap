// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`isCamelCase 1`] = `
[
  {
    "isValid": true,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isCapitalCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": true,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isConstantCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": true,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isDotCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": true,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isKebabCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": true,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isNoCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": true,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isPascalCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": true,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isPascalSnakeCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": true,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isPathCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": true,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isSentenceCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": true,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isSnakeCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": true,
    "value": "snake_case",
  },
  {
    "isValid": false,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;

exports[`isTrainCase 1`] = `
[
  {
    "isValid": false,
    "value": "camelCase",
  },
  {
    "isValid": false,
    "value": "kebab-case",
  },
  {
    "isValid": false,
    "value": "PascalCase",
  },
  {
    "isValid": false,
    "value": "snake_case",
  },
  {
    "isValid": true,
    "value": "Train-Case",
  },
  {
    "isValid": false,
    "value": "Capital Case",
  },
  {
    "isValid": false,
    "value": "CONSTANT_CASE",
  },
  {
    "isValid": false,
    "value": "dot.case",
  },
  {
    "isValid": false,
    "value": "no case",
  },
  {
    "isValid": false,
    "value": "Pascal_Snake_Case",
  },
  {
    "isValid": false,
    "value": "path/case",
  },
  {
    "isValid": false,
    "value": "Sentence case",
  },
  {
    "isValid": false,
    "value": "",
  },
]
`;
